src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -1230,22 +1248,30 @@ inline sycl::event iamin(const char* func_name, Func func, sycl::queue& queue, i
     done.wait();
     if (result_on_device) {
         auto last_ev = queue.submit([&](sycl::handler& cgh) {
```

Comment:
Similar to the buffer case, it would be better to skip this queue submission entirely if `base == index_base::one`.